bentoml/_internal/frameworks/tensorflow_v2.py

```diff
@@ -386,129 +188,166 @@ def __call__(self, inputs):
         labels=labels,
         custom_objects=custom_objects,
         metadata=metadata,
+        signatures=signatures,
     ) as _model:
 
-        if isinstance(model, (str, bytes, os.PathLike, pathlib.Path)):  # type: ignore[reportUnknownMemberType]
-            assert os.path.isdir(model)
-            copy_tree(str(model), _model.path)
-        else:
-            if options:
-                logger.warning(
-                    f"Parameter 'options: {str(options)}' is ignored when "
-                    f"using tensorflow {get_tf_version()}"
-                )
-            tf.saved_model.save(
-                model, _model.path, signatures=signatures, options=options
-            )
-
-        return _model.tag
-
-
-class _TensorflowRunner(BaseModelRunner):
-    def __init__(
-        self,
-        tag: t.Union[str, Tag],
-        predict_fn_name: str,
-        device_id: str,
-        partial_kwargs: t.Optional[t.Dict[str, t.Any]],
-        name: t.Optional[str] = None,
-    ):
-        super().__init__(tag, name=name)
-        self._device_id = device_id
-        self._configure(device_id)
-        self._predict_fn_name = predict_fn_name
-        self._partial_kwargs: t.Dict[str, t.Any] = (
-            partial_kwargs if partial_kwargs is not None else dict()
-        )
-
-    def _configure(self, device_id: str) -> None:
-        if "GPU" in device_id:
-            tf.config.set_visible_devices(device_id, "GPU")
-        self._config_proto = dict(
-            allow_soft_placement=True,
-            log_device_placement=False,
-            intra_op_parallelism_threads=self._num_threads,
-            inter_op_parallelism_threads=self._num_threads,
+        tf.saved_model.save(
+            model,
+            _model.path,
+            signatures=tf_signatures,
+            options=tf_save_options,
```

Comment:
let's avoid mixing the concept of tensorflow's save_options and bentoml's ModelOptions.

I'd recommend creating a TensorflowOptions class from the ModelOptions, and it contains a `save_options` field. This allow us to extend it for other options in the future more easily.